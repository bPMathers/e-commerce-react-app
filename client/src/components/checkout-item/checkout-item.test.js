import React from 'react';
import { shallow } from 'enzyme';

import { CheckoutItem } from './checkout-item.component';

// why am I putting this outside beforeEach?
let wrapper;
let mockClearItem;
let mockAddItem;
let mockRemoveItem;

beforeEach(() => {
  mockClearItem = jest.fn();
  mockAddItem = jest.fn();
  mockRemoveItem = jest.fn();
  const mockProps = {
    cartItem: {
      imageUrl: 'www.imageUrl.com',
      price: 10,
      name: 'itemName',
      quantity: 1
    },
    clearItem: mockClearItem,
    addItem: mockAddItem,
    removeItem: mockRemoveItem
  };

  wrapper = shallow(<CheckoutItem {...mockProps} />);
});

describe('CheckoutItem component', () => {
  it('should render CheckoutItem component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
