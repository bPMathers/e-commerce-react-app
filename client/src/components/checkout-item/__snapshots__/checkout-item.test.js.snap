// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutItem component should render CheckoutItem component 1`] = `
<styled.div>
  <styled.div>
    <img
      alt="item"
      src="www.imageUrl.com"
    />
  </styled.div>
  <styled.span>
    itemName
  </styled.span>
  <QuantityContainer>
    <div
      onClick={[Function]}
    >
      ❮
    </div>
    <span>
      1
    </span>
    <div
      onClick={[Function]}
    >
      ❯
    </div>
  </QuantityContainer>
  <styled.span>
    10
  </styled.span>
  <RemoveButtonContainer
    onClick={[Function]}
  >
    ✕
  </RemoveButtonContainer>
</styled.div>
`;
